### ISSUE TITLE
The `vpgl_algo_test_rational_adjust` test fails under Address Sanitizer (ASan)

### ISSUE THREAD

**Issue by seanm** [vxl/vxl#858]
In current git master the `vpgl_algo_test_rational_adjust` test fails under Address Sanitizer (ASan).  The function `vnl_levenberg_marquardt::lmdif_lsqfun` is reading past of end of a buffer its given.

Here is a dump of my lldb session:

```
$ lldb -- /Users/sean/external/vxl-bin/bin/vpgl_algo_test_all "test_rational_adjust"

(lldb) target create "/Users/sean/external/vxl-bin/bin/vpgl_algo_test_all"
Current executable set to '/Users/sean/external/vxl-bin/bin/vpgl_algo_test_all' (x86_64).
(lldb) settings set -- target.run-args  "test_rational_adjust"

(lldb) r
Process 26297 launched: '/Users/sean/external/vxl-bin/bin/vpgl_algo_test_all' (x86_64)
-----------------------------------------------------------------------------
Start Testing test_rational_adjust:
-----------------------------------------------------------------------------
lmdif: iter 0 err [44.3544, 33.3433, 37.6577, =================================================================
==26297==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6030000006b8 at pc 0x000100823937 bp 0x7ffeefbfab90 sp 0x7ffeefbfab88
READ of size 8 at 0x6030000006b8 thread T0
2021-09-16 16:28:03.221815-0400 atos[26300:27985143] examining /Users/USER/*/vpgl_algo_test_all [26297]
2021-09-16 16:28:03.432997-0400 atos[26301:27985148] examining /Users/USER/*/vpgl_algo_test_all [26297]
    #0 0x100823936 in vnl_levenberg_marquardt::lmdif_lsqfun(long*, long*, double*, double*, long*, void*) vnl_levenberg_marquardt.cxx:84
    #1 0x102959a63 in v3p_netlib_lmdif_ lmdif.c:378
    #2 0x10082aef4 in vnl_levenberg_marquardt::minimize_without_gradient(vnl_vector<double>&) vnl_levenberg_marquardt.cxx:171
    #3 0x1008206fb in vnl_levenberg_marquardt::minimize(vnl_vector<double>&) vnl_levenberg_marquardt.cxx:118
    #4 0x1003e66db in vpgl_rational_adjust::adjust(vpgl_rational_camera<double> const&, std::__1::vector<vgl_point_2d<double>, std::__1::allocator<vgl_point_2d<double> > >, std::__1::vector<vgl_point_3d<double>, std::__1::allocator<vgl_point_3d<double> > >, vpgl_rational_camera<double>&) vpgl_rational_adjust.cxx:169
    #5 0x100106d78 in test_rational_adjust() test_rational_adjust.cxx:121
    #6 0x100105c65 in test_rational_adjust_main(int, char**) test_rational_adjust.cxx:127
    #7 0x102241078 in testlib_run_test_unit(unsigned long, int, char**) testlib_main.cxx:112
    #8 0x102241bb5 in testlib_main(int, char**) testlib_main.cxx:154
    #9 0x1000038c3 in main test_driver.cxx:55
    #10 0x7fff64b663d4 in start (libdyld.dylib:x86_64+0x163d4)

0x6030000006b8 is located 0 bytes to the right of 24-byte region [0x6030000006a0,0x6030000006b8)
allocated by thread T0 here:
    #0 0x105937ee2 in wrap__Znam (libclang_rt.asan_osx_dynamic.dylib:x86_64+0x6bee2)
    #1 0x1009b84d9 in vnl_sse_alloc(unsigned long, unsigned int) vnl_sse.h:106
    #2 0x1009b7e10 in vnl_c_vector_alloc(unsigned long, unsigned int) vnl_c_vector.hxx:350
    #3 0x1009b7e7d in vnl_c_vector<double>::allocate_T(unsigned long) vnl_c_vector.hxx:404
    #4 0x10151bdf0 in vnl_vector<double>::vnl_vector(unsigned long) vnl_vector.hxx:85
    #5 0x10151bf75 in vnl_vector<double>::vnl_vector(unsigned long) vnl_vector.hxx:84
    #6 0x1003e6227 in vpgl_rational_adjust::adjust(vpgl_rational_camera<double> const&, std::__1::vector<vgl_point_2d<double>, std::__1::allocator<vgl_point_2d<double> > >, std::__1::vector<vgl_point_3d<double>, std::__1::allocator<vgl_point_3d<double> > >, vpgl_rational_camera<double>&) vpgl_rational_adjust.cxx:154
    #7 0x100106d78 in test_rational_adjust() test_rational_adjust.cxx:121
    #8 0x100105c65 in test_rational_adjust_main(int, char**) test_rational_adjust.cxx:127
    #9 0x102241078 in testlib_run_test_unit(unsigned long, int, char**) testlib_main.cxx:112
    #10 0x102241bb5 in testlib_main(int, char**) testlib_main.cxx:154
    #11 0x1000038c3 in main test_driver.cxx:55
    #12 0x7fff64b663d4 in start (libdyld.dylib:x86_64+0x163d4)

SUMMARY: AddressSanitizer: heap-buffer-overflow vnl_levenberg_marquardt.cxx:84 in vnl_levenberg_marquardt::lmdif_lsqfun(long*, long*, double*, double*, long*, void*)
Shadow bytes around the buggy address:
  0x1c0600000080: 00 00 00 00 fa fa fd fd fd fd fa fa 00 00 00 00
  0x1c0600000090: fa fa fd fd fd fd fa fa 00 00 00 00 fa fa fd fd
  0x1c06000000a0: fd fd fa fa 00 00 00 00 fa fa fd fd fd fd fa fa
  0x1c06000000b0: 00 00 00 00 fa fa fd fd fd fa fa fa fd fd fd fd
  0x1c06000000c0: fa fa 00 00 00 fa fa fa 00 00 00 fa fa fa 00 00
=>0x1c06000000d0: 00 fa fa fa 00 00 00[fa]fa fa 00 00 00 fa fa fa
  0x1c06000000e0: 00 00 00 fa fa fa 00 00 00 fa fa fa 00 00 00 fa
  0x1c06000000f0: fa fa 00 00 00 fa fa fa 00 00 00 00 fa fa fa fa
  0x1c0600000100: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x1c0600000110: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
  0x1c0600000120: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
2021-09-16 16:28:09.432971-0400 vpgl_algo_test_all[26297:27985119] =================================================================
2021-09-16 16:28:09.433772-0400 vpgl_algo_test_all[26297:27985119] ==26297==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x6030000006b8 at pc 0x000100823937 bp 0x7ffeefbfab90 sp 0x7ffeefbfab88
2021-09-16 16:28:09.433808-0400 vpgl_algo_test_all[26297:27985119] READ of size 8 at 0x6030000006b8 thread T0
2021-09-16 16:28:09.433827-0400 vpgl_algo_test_all[26297:27985119]     #0 0x100823936 in vnl_levenberg_marquardt::lmdif_lsqfun(long*, long*, double*, double*, long*, void*) vnl_levenberg_marquardt.cxx:84
2021-09-16 16:28:09.433843-0400 vpgl_algo_test_all[26297:27985119]     #1 0x102959a63 in v3p_netlib_lmdif_ lmdif.c:378
2021-09-16 16:28:09.433864-0400 vpgl_algo_test_all[26297:27985119]     #2 0x10082aef4 in vnl_levenberg_marquardt::minimize_without_gradient(vnl_vector<double>&) vnl_levenberg_marquardt.cxx:171
2021-09-16 16:28:09.433880-0400 vpgl_algo_test_all[26297:27985119]     #3 0x1008206fb in vnl_levenberg_marquardt::minimize(vnl_vector<double>&) vnl_levenberg_marquardt.cxx:118
2021-09-16 16:28:09.433900-0400 vpgl_algo_test_all[26297:27985119]     #4 0x1003e66db in vpgl_rational_adjust::adjust(vpgl_rational_camera<double> const&, std::__1::vector<vgl_point_2d<double>, std::__1::allocator<vgl_point_2d<double> > >, std::__1::vector<vgl_point_3d<double>, std::__1::allocator<vgl_point_3d<double> > >, vpgl_rational_camera<double>&) vpgl_rational_adjust.cxx:169
2021-09-16 16:28:09.433918-0400 vpgl_algo_test_all[26297:27985119]     #5 0x100106d78 in test_rational_adjust() test_rational_adjust.cxx:121
2021-09-16 16:28:09.433938-0400 vpgl_algo_test_all[26297:27985119]     #6 0x100105c65 in test_rational_adjust_main(int, char**) test_rational_adjust.cxx:127
2021-09-16 16:28:09.433968-0400 vpgl_algo_test_all[26297:27985119]     #7 0x102241078 in testlib_run_test_unit(unsigned long, int, char**) testlib_main.cxx:112
2021-09-16 16:28:09.433987-0400 vpgl_algo_test_all[26297:27985119]     #8 0x102241bb5 in testlib_main(int, char**) testlib_main.cxx:154
2021-09-16 16:28:09.434003-0400 vpgl_algo_test_all[26297:27985119]     #9 0x1000038c3 in main test_driver.cxx:55
2021-09-16 16:28:09.434018-0400 vpgl_algo_test_all[26297:27985119]     #10 0x7fff64b663d4 in start (libdyld.dylib:x86_64+0x163d4)
2021-09-16 16:28:09.434032-0400 vpgl_algo_test_all[26297:27985119] 
2021-09-16 16:28:09.434046-0400 vpgl_algo_test_all[26297:27985119] 0x6030000006b8 is located 0 bytes to the right of 24-byte region [0x6030000006a0,0x6030000006b8)
2021-09-16 16:28:09.434061-0400 vpgl_algo_test_all[26297:27985119] allocated by thread T0 here:
2021-09-16 16:28:09.434076-0400 vpgl_algo_test_all[26297:27985119]     #0 0x105937ee2 in wrap__Znam (libclang_rt.asan_osx_dynamic.dylib:x86_64+0x6bee2)
2021-09-16 16:28:09.434092-0400 vpgl_algo_test_all[26297:27985119]     #1 0x1009b84d9 in vnl_sse_alloc(unsigned long, unsigned int) vnl_sse.h:106
2021-09-16 16:28:09.434109-0400 vpgl_algo_test_all[26297:27985119]     #2 0x1009b7e10 in vnl_c_vector_alloc(unsigned long, unsigned int) vnl_c_vector.hxx:350
2021-09-16 16:28:09.434125-0400 vpgl_algo_test_all[26297:27985119]     #3 0x1009b7e7d in vnl_c_vector<double>::allocate_T(unsigned long) vnl_c_vector.hxx:404
2021-09-16 16:28:09.434140-0400 vpgl_algo_test_all[26297:27985119]     #4 0x10151bdf0 in vnl_vector<double>::vnl_vector(unsigned long) vnl_vector.hxx:85
2021-09-16 16:28:09.434155-0400 vpgl_algo_test_all[26297:27985119]     #5 0x10151bf75 in vnl_vector<double>::vnl_vector(unsigned long) vnl_vector.hxx:84
2021-09-16 16:28:09.434173-0400 vpgl_algo_test_all[26297:27985119]     #6 0x1003e6227 in vpgl_rational_adjust::adjust(vpgl_rational_camera<double> const&, std::__1::vector<vgl_point_2d<double>, std::__1::allocator<vgl_point_2d<double> > >, std::__1::vector<vgl_point_3d<double>, std::__1::allocator<vgl_point_3d<double> > >, vpgl_rational_camera<double>&) vpgl_rational_adjust.cxx:154
2021-09-16 16:28:09.434190-0400 vpgl_algo_test_all[26297:27985119]     #7 0x100106d78 in test_rational_adjust() test_rational_adjust.cxx:121
2021-09-16 16:28:09.434209-0400 vpgl_algo_test_all[26297:27985119]     #8 0x100105c65 in test_rational_adjust_main(int, char**) test_rational_adjust.cxx:127
2021-09-16 16:28:09.434224-0400 vpgl_algo_test_all[26297:27985119]     #9 0x102241078 in testlib_run_test_unit(unsigned long, int, char**) testlib_main.cxx:112
2021-09-16 16:28:09.434260-0400 vpgl_algo_test_all[26297:27985119]     #10 0x102241bb5 in testlib_main(int, char**) testlib_main.cxx:154
2021-09-16 16:28:09.434282-0400 vpgl_algo_test_all[26297:27985119]     #11 0x1000038c3 in main test_driver.cxx:55
2021-09-16 16:28:09.434298-0400 vpgl_algo_test_all[26297:27985119]     #12 0x7fff64b663d4 in start (libdyld.dylib:x86_64+0x163d4)
2021-09-16 16:28:09.434315-0400 vpgl_algo_test_all[26297:27985119] 
2021-09-16 16:28:09.434328-0400 vpgl_algo_test_all[26297:27985119] SUMMARY: AddressSanitizer: heap-buffer-overflow vnl_levenberg_marquardt.cxx:84 in vnl_levenberg_marquardt::lmdif_lsqfun(long*, long*, double*, double*, long*, void*)
2021-09-16 16:28:09.434346-0400 vpgl_algo_test_all[26297:27985119] Shadow bytes around the buggy address:
2021-09-16 16:28:09.434361-0400 vpgl_algo_test_all[26297:27985119]   0x1c0600000080: 00 00 00 00 fa fa fd fd fd fd fa fa 00 00 00 00
2021-09-16 16:28:09.434376-0400 vpgl_algo_test_all[26297:27985119]   0x1c0600000090: fa fa fd fd fd fd fa fa 00 00 00 00 fa fa fd fd
2021-09-16 16:28:09.434393-0400 vpgl_algo_test_all[26297:27985119]   0x1c06000000a0: fd fd fa fa 00 00 00 00 fa fa fd fd fd fd fa fa
2021-09-16 16:28:09.434408-0400 vpgl_algo_test_all[26297:27985119]   0x1c06000000b0: 00 00 00 00 fa fa fd fd fd fa fa fa fd fd fd fd
2021-09-16 16:28:09.434424-0400 vpgl_algo_test_all[26297:27985119]   0x1c06000000c0: fa fa 00 00 00 fa fa fa 00 00 00 fa fa fa 00 00
2021-09-16 16:28:09.434444-0400 vpgl_algo_test_all[26297:27985119] =>0x1c06000000d0: 00 fa fa fa 00 00 00[fa]fa fa 00 00 00 fa fa fa
2021-09-16 16:28:09.434462-0400 vpgl_algo_test_all[26297:27985119]   0x1c06000000e0: 00 00 00 fa fa fa 00 00 00 fa fa fa 00 00 00 fa
2021-09-16 16:28:09.434479-0400 vpgl_algo_test_all[26297:27985119]   0x1c06000000f0: fa fa 00 00 00 fa fa fa 00 00 00 00 fa fa fa fa
2021-09-16 16:28:09.434495-0400 vpgl_algo_test_all[26297:27985119]   0x1c0600000100: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-09-16 16:28:09.434510-0400 vpgl_algo_test_all[26297:27985119]   0x1c0600000110: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-09-16 16:28:09.434526-0400 vpgl_algo_test_all[26297:27985119]   0x1c0600000120: fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa fa
2021-09-16 16:28:09.434541-0400 vpgl_algo_test_all[26297:27985119] Shadow byte legend (one shadow byte represents 8 application bytes):
2021-09-16 16:28:09.434556-0400 vpgl_algo_test_all[26297:27985119]   Addressable:           00
2021-09-16 16:28:09.434575-0400 vpgl_algo_test_all[26297:27985119]   Partially addressable: 01 02 03 04 05 06 07
2021-09-16 16:28:09.434591-0400 vpgl_algo_test_all[26297:27985119]   Heap left redzone:       fa
2021-09-16 16:28:09.434606-0400 vpgl_algo_test_all[26297:27985119]   Freed heap region:       fd
2021-09-16 16:28:09.434623-0400 vpgl_algo_test_all[26297:27985119]   Stack left redzone:      f1
2021-09-16 16:28:09.434638-0400 vpgl_algo_test_all[26297:27985119]   Stack mid redzone:       f2
2021-09-16 16:28:09.434652-0400 vpgl_algo_test_all[26297:27985119]   Stack right redzone:     f3
2021-09-16 16:28:09.434668-0400 vpgl_algo_test_all[26297:27985119]   Stack after return:      f5
2021-09-16 16:28:09.434683-0400 vpgl_algo_test_all[26297:27985119]   Stack use after scope:   f8
2021-09-16 16:28:09.434697-0400 vpgl_algo_test_all[26297:27985119]   Global redzone:          f9
2021-09-16 16:28:09.434714-0400 vpgl_algo_test_all[26297:27985119]   Global init order:       f6
2021-09-16 16:28:09.434729-0400 vpgl_algo_test_all[26297:27985119]   Poisoned by user:        f7
2021-09-16 16:28:09.434743-0400 vpgl_algo_test_all[26297:27985119]   Container overflow:      fc
2021-09-16 16:28:09.434760-0400 vpgl_algo_test_all[26297:27985119]   Array cookie:            ac
2021-09-16 16:28:09.434777-0400 vpgl_algo_test_all[26297:27985119]   Intra object redzone:    bb
2021-09-16 16:28:09.434794-0400 vpgl_algo_test_all[26297:27985119]   ASan internal:           fe
2021-09-16 16:28:09.434811-0400 vpgl_algo_test_all[26297:27985119]   Left alloca redzone:     ca
2021-09-16 16:28:09.434827-0400 vpgl_algo_test_all[26297:27985119]   Right alloca redzone:    cb
2021-09-16 16:28:09.434843-0400 vpgl_algo_test_all[26297:27985119]   Shadow gap:              cc
==26297==ABORTING
(lldb) AddressSanitizer report breakpoint hit. Use 'thread info -s' to get extended information about the report.
Process 26297 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = Heap buffer overflow
    frame #0: 0x0000000105933900 libclang_rt.asan_osx_dynamic.dylib`__asan::AsanDie()
libclang_rt.asan_osx_dynamic.dylib`__asan::AsanDie:
->  0x105933900 <+0>: pushq  %rbp
    0x105933901 <+1>: movq   %rsp, %rbp
    0x105933904 <+4>: pushq  %rbx
    0x105933905 <+5>: pushq  %rax
Target 0: (vpgl_algo_test_all) stopped.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = Heap buffer overflow
  * frame #0: 0x0000000105933900 libclang_rt.asan_osx_dynamic.dylib`__asan::AsanDie()
    frame #1: 0x00000001059497ff libclang_rt.asan_osx_dynamic.dylib`__sanitizer::Die() + 175
    frame #2: 0x0000000105931829 libclang_rt.asan_osx_dynamic.dylib`__asan::ScopedInErrorReport::~ScopedInErrorReport() + 409
    frame #3: 0x0000000105931033 libclang_rt.asan_osx_dynamic.dylib`__asan::ReportGenericError(unsigned long, unsigned long, unsigned long, unsigned long, bool, unsigned long, unsigned int, bool) + 387
    frame #4: 0x0000000105931d28 libclang_rt.asan_osx_dynamic.dylib`__asan_report_load8 + 40
    frame #5: 0x0000000100823937 vpgl_algo_test_all`vnl_levenberg_marquardt::lmdif_lsqfun(n=0x00007ffeefbfcbc0, p=0x00007ffeefbfcbe0, x=0x00006030000006a0, fx=0x000060d0000001e0, iflag=0x00007ffeefbfb1c0, userdata=0x00007ffeefbfda20) at vnl_levenberg_marquardt.cxx:84:20
    frame #6: 0x0000000102959a64 vpgl_algo_test_all`v3p_netlib_lmdif_(fcn=(vpgl_algo_test_all`vnl_levenberg_marquardt::lmdif_lsqfun(long*, long*, double*, double*, long*, void*) at vnl_levenberg_marquardt.cxx:72), m=0x00007ffeefbfcbc0, n=0x00007ffeefbfcbe0, x=0x0000603000000698, fvec=0x000060d0000001d8, ftol=0x00007ffeefbfda38, xtol=0x00007ffeefbfda28, gtol=0x00007ffeefbfda40, maxfev=0x00007ffeefbfda30, epsfcn=0x00007ffeefbfda48, diag=0x00006030000006c8, mode=0x00007ffeefbfccc0, factor=0x00007ffeefbfcce0, nprint=0x00007ffeefbfcd00, info=0x00007ffeefbfcf00, nfev=0x00007ffeefbfda58, fjac=0x0000613fffffffa8, ldfjac=0x00007ffeefbfcbc0, ipvt=0x0000603000000638, qtf=0x00006030000006f8, wa1=0x0000603000000728, wa2=0x0000603000000758, wa3=0x0000603000000788, wa4=0x000060d0000002a8, userdata=0x00007ffeefbfda20) at lmdif.c:378:9
    frame #7: 0x000000010082aef5 vpgl_algo_test_all`vnl_levenberg_marquardt::minimize_without_gradient(this=0x00007ffeefbfda20, x=0x00007ffeefbfdb30) at vnl_levenberg_marquardt.cxx:171:3
    frame #8: 0x00000001008206fc vpgl_algo_test_all`vnl_levenberg_marquardt::minimize(this=0x00007ffeefbfda20, x=0x00007ffeefbfdb30) at vnl_levenberg_marquardt.cxx:118:12
    frame #9: 0x00000001003e66dc vpgl_algo_test_all`vpgl_rational_adjust::adjust(initial_rcam=0x00007ffeefbfe3e0, img_pts=size=9, geo_pts=size=9, adj_rcam=0x00007ffeefbfeaf0) at vpgl_rational_adjust.cxx:169:11
    frame #10: 0x0000000100106d79 vpgl_algo_test_all`test_rational_adjust() at test_rational_adjust.cxx:121:3
    frame #11: 0x0000000100105c66 vpgl_algo_test_all`test_rational_adjust_main((null)=1, (null)=0x00007ffeefbff798) at test_rational_adjust.cxx:127:1
    frame #12: 0x0000000102241079 vpgl_algo_test_all`testlib_run_test_unit(i=11, argc=1, argv=0x00007ffeefbff798) at testlib_main.cxx:112:10
    frame #13: 0x0000000102241bb6 vpgl_algo_test_all`testlib_main(argc=2, argv=0x00007ffeefbff790) at testlib_main.cxx:154:16
    frame #14: 0x00000001000038c4 vpgl_algo_test_all`main(argc=2, argv=0x00007ffeefbff790) at test_driver.cxx:55:1
    frame #15: 0x00007fff64b663d5 libdyld.dylib`start + 1
(lldb) fr sel 5
frame #5: 0x0000000100823937 vpgl_algo_test_all`vnl_levenberg_marquardt::lmdif_lsqfun(n=0x00007ffeefbfcbc0, p=0x00007ffeefbfcbe0, x=0x00006030000006a0, fx=0x000060d0000001e0, iflag=0x00007ffeefbfb1c0, userdata=0x00007ffeefbfda20) at vnl_levenberg_marquardt.cxx:84:20
   81  	  {
   82  	    if (self->trace)
   83  	      std::cerr << "lmdif: iter " << self->num_iterations_ << " err [" << x[0] << ", " << x[1] << ", " << x[2] << ", "
-> 84  	                << x[3] << ", " << x[4] << ", ... ] = " << ref_fx.magnitude() << '\n';
   85  	
   86  	    f->trace(self->num_iterations_, ref_x, ref_fx);
   87  	    ++(self->num_iterations_);

(lldb) p *n
(long) $2 = 18

(lldb) p *p
(long) $3 = 3
```

**Comment by drewgilliam** [comment]
Traceback is quoting this
https://github.com/vxl/vxl/blob/master/core/vnl/algo/vnl_levenberg_marquardt.cxx#L83-L84
```
   83  	      std::cerr << "lmdif: iter " << self->num_iterations_ << " err [" << x[0] << ", " << x[1] << ", " << x[2] << ", "
-> 84  	                << x[3] << ", " << x[4] << ", ... ] = " << ref_fx.magnitude() << '\n';
```

The input argument states `double * x,   // I   Solution vector, size n`, so x isn't necessarily of size 5.  That `std::cerr` might be printing garbage in this case if `self->trace` was active.

Not sure if you'd see the same issue here
https://github.com/vxl/vxl/blob/master/core/vnl/algo/vnl_levenberg_marquardt.cxx#L243-L244

Note that prior to a recent style change, the last time these lines were touched looks like it was 20 years ago...

**Comment by seanm** [comment]
>The input argument states double * x, // I Solution vector, size n, so x isn't necessarily of size 5.

Indeed, it's definitely smaller, which is what ASan is telling us. :) Note though in my lldb output I printed that `n` is 18. Looking closer, ASan also prints the backtrace of the allocation, and its an allocation of 3 doubles (vpgl_rational_adjust.cxx:154), so this code is seemingly wrong to assume there are at least 5. But again, `n` is 18, not 3, so that's odd... digging further, I found the 3!  That comment is just wrong: `x` is not size `n`, it is size `p`!  And the only thing wrong is the logging code assuming there are at least 5.

>Not sure if you'd see the same issue here...

Probably!  But seems no test exercises it being less that 5.

Patch coming...
